python/python/lance/dataset.py

```diff
@@ -1679,16 +1691,30 @@ def create_index(
                     train_ivf_centroids_on_accelerator,
                 )
 
+                times = []
+                times.append(time.time())
                 ivf_centroids, kmeans = train_ivf_centroids_on_accelerator(
                     self,
                     column[0],
                     num_partitions,
                     metric,
                     accelerator,
                 )
+                times.append(time.time())
+                logging.info("ivf training time: %ss", times[1] - times[0])
+                num_sub_vectors_cur = None
+                if "PQ" in index_type and pq_codebook is None:
+                    # compute residual subspace columns in the same pass
+                    num_sub_vectors_cur = num_sub_vectors
                 partitions_file = compute_partitions(
-                    self, column[0], kmeans, batch_size=20480
+                    self,
+                    column[0],
+                    kmeans,
+                    batch_size=20480,
```

Comment:
nit: maybe this should be adaptive to dimension/fraction of GPU memory aval

Reply:
Opened a new issue for this #2978